packages/base/src/3dview/mainview.tsx

```diff
@@ -1514,7 +1506,13 @@ export class MainView extends React.Component<IProps, IStates> {
           factor: this._explodedView.factor
         });
 
-        group.position.set(0, 0, 0);
+        group.position.copy(
+          new THREE.Vector3(
+            positionArray[0],
+            positionArray[1],
+            positionArray[2]
+          )
+        );
         group.translateOnAxis(explodedState.vector, explodedState.distance);
```

Comment:
Try:

```suggestion
        group.position.copy(
          new THREE.Vector3(
            positionArray[0] + explodedState.vector.x * explodedState.distance,
            positionArray[1] + explodedState.vector.y * explodedState.distance,
            positionArray[2] + explodedState.vector.z * explodedState.distance          
          )
        );
```

It seems `translateOnAxis` behaves weirdly with rotated shapes

Reply:
For the record, this is because `translateOnAxis` considers the axis in the Object space, we want it in the world space 

https://threejs.org/docs/index.html?q=object#api/en/core/Object3D.translateOnAxis